homeassistant/components/switchbot_cloud/climate.py

```diff
@@ -83,22 +90,51 @@ class SwitchBotCloudAirConditioner(SwitchBotCloudEntity, ClimateEntity):
     _attr_precision = 1
     _attr_name = None
 
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added."""
+        await super().async_added_to_hass()
+
+        if not (last_state := await self.async_get_last_state()):
+            return
+        _LOGGER.debug("Last state attributes: %s", last_state.attributes)
+        self._attr_hvac_mode = HVACMode(last_state.state)
+        self._attr_fan_mode = last_state.attributes.get(
+            ATTR_FAN_MODE, self._attr_fan_mode
+        )
+        self._attr_target_temperature = last_state.attributes.get(
+            ATTR_TEMPERATURE, self._attr_target_temperature
+        )
+
+    def _get_mode(self, hvac_mode: HVACMode | None = None) -> int:
```

Comment:
The default value is not needed
```suggestion
    def _get_mode(self, hvac_mode: HVACMode | None) -> int:
```